pkg/abaputils/manageGitRepositoryUtils.go

```diff
@@ -210,6 +211,19 @@ func (repo *Repository) GetRequestBodyForCommitOrTag() (requestBodyString string
 	return requestBodyString
 }
 
+func (repo *Repository) GetRequestBodyForBYOGCredentials() (byogBodyString string) {
+	if repo.ByogAuthMethod != "" {
+		byogBodyString += `, "auth_method":"` + repo.ByogAuthMethod + `"`
+	}
+	if repo.ByogUsername != "" {
```

Comment:
If ByogUsername is provided, and ByogPassword is missing, only half the request body is created.

How could we handle such situations? Omitting the request body? Throwing an error message?